synapse/tests/test_lib_layer.py

```diff
@@ -52,6 +52,19 @@ class LayerTest(s_t_utils.SynTest):
                     await liftByHandler('univ:ival', 1)
                     await liftByHandler('form:ival', 1)
 
+    async def test_layer_buidcache(self):
+        async with self.getTestCore() as core:
+            async with await core.snap() as snap:
+                node = await snap.addNode('test:str', 'a', {'tick': '1970'})
+                await node.addTag('foo')
+
+            # new snap -> no cached buids in snap
+            async with await core.snap() as snap:
+                nodes = await alist(snap.getNodesBy('test:str'))
+                self.len(1, nodes)
+                node = nodes[0]
+                self.eq(node.tags, {'foo': (None, None)})
```

Comment:
would be good to see that, inbetween snaps, the buidcache on the cortex's view.layers[0] is being modified in flight; as well as the deletion case ensuring that we cannot retrieve deleted nodes out of the layer buid cache.

Reply:
done